manifests/init.pp

```diff
@@ -231,7 +231,7 @@
       path    => "${confdir}/puppet.conf",
       section => 'main',
       setting => 'hiera_config',
-      value   => $hiera_yaml,
+      value   => '$confdir/hiera.yaml',
```

Comment:
Let's take that up on irc or in another issue - I'm using this with Puppet 4 just fine, and have been for a while.